lib/sycamore/sycamore/transforms/partition.py

```diff
@@ -490,6 +492,83 @@ def partition(self, document: Document) -> Document:
         return document
 
 
+class ArynPartitioner(Partitioner):
+    """
+    The ArynPartitioner runs a SycamorePartitioner on Aryn's GPU servers to offload compute.
+
+    Args:
+        aryn_token: The account token used to authenticate with Aryn's servers.
+        threshold: The threshold to use for accepting the models predicted bounding boxes. A lower
+             value will include more objects, but may have overlaps, a higher value will reduce the
+             number of overlaps, but may miss legitimate objects.
+        use_ocr: Whether to use OCR to extract text from the PDF. If false, we will attempt to extract
+             the text from the underlying PDF.
+        ocr_images: If set with use_ocr, will attempt to OCR regions of the document identified as images.
+        ocr_tables: If set with use_ocr, will attempt to OCR regions on the document identified as tables.
+             Should not be set when `extract_table_structure` is true.
+        extract_table_structure: If true, runs a separate table extraction model to extract cells from
+             regions of the document identified as tables.
+        extract_images: If true, crops each region identified as an image and attaches it to the associated
+             ImageElement. This can later be fed into the SummarizeImages transform.
+
+    Example:
+        The following shows an example of using the ArynPartitioner to partition a PDF and extract
+        both table structure and images.
+
+        .. code-block:: python
+
+            context = sycamore.init()
+            partitioner = ArynPartitioner(extract_table_structure=True, extract_images=True)
+            context.read.binary(paths, binary_format="pdf")\
+                .partition(partitioner=partitioner)
+    """
+
+    def __init__(
+        self,
+        aryn_token: str,
+        threshold: float = 0.4,
+        use_ocr: bool = False,
+        ocr_images: bool = False,
+        ocr_tables: bool = False,
+        extract_table_structure: bool = False,
+        extract_images: bool = False,
+        aryn_partitioner_address: str = _DEFAULT_ARYN_PARTITIONER_ADDRESS,
+    ):
+        super().__init__(device="cpu", batch_size=1)
+        self._aryn_token = aryn_token
+        self._threshold = threshold
+        self._use_ocr = use_ocr
+        self._ocr_images = ocr_images
+        self._ocr_tables = ocr_tables
+        self._extract_table_structure = extract_table_structure
+        self._extract_images = extract_images
+        self._aryn_partitioner_address = aryn_partitioner_address
+
+    def partition(self, document: Document):
+        binary = io.BytesIO(document.data["binary_representation"])
```

Comment:
Minor: I would prefer `document.binary_representation` rather than accessing the internal dictionary directly. That way any other code we put in the property accessor is invoked.